tools/lib/test_server.py

```diff
@@ -54,10 +54,11 @@ def test_server_running(force: bool=False, external_host: str='testserver',
     log = sys.stdout
     if log_file:
         if os.path.exists(log_file) and os.path.getsize(log_file) < 100000:
-            log = open(log_file, 'a')
-            log.write('\n\n')
+            with open(log_file, 'a') as f:
+                f.write('\n\n')
         else:
-            log = open(log_file, 'w')
+            with open(log_file, 'w') as f:
+                f.close()
```

Comment:
So, this change will completely break the logging here.  I know this feels like a mechanical migration, but it should have been obvious that there was something interesting going on here, given that opening a file to immediately close it doesn't make any sense.  It's really important when doing a mechanical-feeling migration to actually read the code and think about it -- a lot of nasty and expensive to track down bugs result from this kind of error in a big refactor that completely changes the meaning of the original code.

Read carefully: this `log` variable is passed into other functions to control where data is logged to.

Reply:
Right, that was a big error on my part.  I'll slow down, and really think about what I'm doing, even in mechanical tasks such as this.